src/wallet/wallet.cpp

```diff
@@ -3129,7 +3151,11 @@ bool CReserveKey::GetReservedKey(CPubKey& pubkey, bool internal)
         if (nIndex != -1)
             vchPubKey = keypool.vchPubKey;
         else {
-            return false;
+            if (pwallet->IsLocked())
```

Comment:
Right. This is no longer required because we always derive at minimum 2 internal keys. And `ReserveKeyFromKeyPool` will topup the keypool.
Will remove that part.